### PR TITLE
chore: allow teams to add/remove resources & data sources

### PR DESCRIPTION
**Pull request by ctreatma**
This refactors the SDKv2 and framework providers so that the data sources and resources for each Equinix service are defined in separate files.  This enables each team to add or remove their own data sources and resources without requiring review from DRE.

### CODE REVIEW

**Review by displague — COMMENTED** [review]
Should the new internal/provider/services files be named/routed in https://github.com/equinix/terraform-provider-equinix/blob/main/CODEOWNERS? Specifically or through wildcard?

**Comment by ctreatma** [comment]
The code ownership mapping is address by #804 (although that PR was opened for a different reason).

**Review by displague — APPROVED** [review]

**Comment by ctreatma** [comment]
One member of each of @equinix/governor-digin-fabric and @equinix/governor-ne-network-edge-engineering will need to approve this before it can be merged.

**Review by kpdhulipala — APPROVED** [review]

**Review by srushti-patl — APPROVED** [review]

**Pull request merged**
